hardware/system.py

```diff
@@ -0,0 +1,223 @@
+                        hw_lst.append(
+                            ('network', name.text, 'ipv4-netmask', netmask))
+                        cidr = detect_utils.get_cidr(netmask)
+                        hw_lst.append(
+                            ('network', name.text, 'ipv4-cidr', cidr))
+                        net = (ipaddress.IPv4Interface('%s/%s' % (ipv4, cidr))
+                               .network.network_address)
+                        hw_lst.append(
+                            ('network', name.text, 'ipv4-network', str(net)))
+                    except Exception as excpt:
+                        sys.stderr.write('unable to get info for %s: %s\n'
+                                         % (name.text, str(excpt)))
+
+                _find_element(elt, "configuration/setting[@id='link']", 'link',
+                              name.text, 'network', 'value')
+                _find_element(elt, "configuration/setting[@id='driver']",
+                              'driver', name.text, 'network', 'value')
+                _find_element(elt, "configuration/setting[@id='duplex']",
+                              'duplex', name.text, 'network', 'value')
+                _find_element(elt, "configuration/setting[@id='speed']",
+                              'speed', name.text, 'network', 'value')
+                _find_element(elt, "configuration/setting[@id='latency']",
+                              'latency', name.text, 'network', 'value')
+                _find_element(elt,
+                              "configuration/setting[@id='autonegotiation']",
+                              'autonegotiation', name.text, 'network', 'value')
+
+                # lshw is not able to get the complete mac addr for ib
+                # devices Let's workaround it with an ip command.
+                if name.text.startswith('ib'):
+                    cmds = "ip addr show %s | grep link | awk '{print $2}'"
+                    status_ip, output_ip = detect_utils.cmd(cmds % name.text)
+                    if status_ip == 0:
+                        hw_lst.append(('network',
+                                       name.text,
+                                       'serial',
+                                       output_ip.split('\n')[0].lower()))
+                else:
+                    _find_element(elt, 'serial', 'serial', name.text,
+                                  'network', transform=lambda x: x.lower())
+
+                if not nic_id:
+                    nic_id = detect_utils.get_value(
+                        hw_lst, 'network', name.text, 'serial')
+                    nic_id = nic_id.replace(':', '')
+
+                detect_utils.get_ethtool_status(hw_lst, name.text)
+                detect_utils.get_lld_status(hw_lst, name.text)
+
+        detect_utils.fix_bad_serial(hw_lst, uuid, mobo_id, nic_id)
+
+    else:
+        sys.stderr.write("Unable to run lshw: %s\n" % output)
+        return False
```

Comment:
We should probably return [] here to stay consistent about the function output

Reply:
whoops I didn't see this, thanks!